rust/src/enip/enip.rs

```diff
@@ -272,7 +269,7 @@ impl EnipState {
         }
     }
     fn parse_tcp(
-        &mut self, stream_slice: StreamSlice, request: bool, flow: *const Flow,
+        &mut self, stream_slice: StreamSlice, request: bool, flow: *const Flow, dir: Direction,
```

Comment:
You should merge the `request` and `dir` arguments ;-)

Reply:
Indeed that's better. Thanks :)